File: heron/lc246.java
public class Solution {
    public boolean isStrobogrammatic(String num) {

        Map<Character, Character> map = new HashMap<>();
        map.put('6', '9');
        map.put('9', '6');
        map.put('8', '8');
        map.put('0', '0');
        map.put('1', '1');
        Stack<Character> stack = new Stack<>();
        for (Character c : num.toCharArray()) {
            stack.push(c);
            System.out.println("" + c);
        }
        for (Character c : num.toCharArray()) {
            char sc = stack.pop();
            System.out.println("" + c + ", " + sc);
            if (!map.containsKey(c) || map.get(c) != sc) {
                return false;
            }
        }

        return stack.isEmpty();
    }
}

File: alice/LC246.py
class Solution(object):
    def isStrobogrammatic(self, num):
        """
        :type num: str
        :rtype: bool
        """
        
        mapping = {'0': '0', '1': '1', '6': '9', '8': '8', '9': '6'}

        temp = ''
        for x in num:
            if x in mapping:
                temp += mapping[x]
            else:
                return False

        return num == temp[::-1]